### PR TITLE
Update User Guide for First Skill Experience

### PR DESCRIPTION
**Pull request by daltonhuynh**
## Description

* Moved "How does it work" to "Developing with Controls"
* Moved Controls source sections to "Contributing"
* Added "Purpose of Guide" section
* Updated content for "Prerequisites" and JS vs TS

## Motivation and Context
Condensing start of guide so developers get to their first skill more quickly.

## License

- [x] I confirm that this pull request can be released under the Apache 2 license

[issues]: https://https://github.com/alexa/ask-sdk-controls/issues
[license]: http://aws.amazon.com/apache2.0/
[cla]: http://en.wikipedia.org/wiki/Contributor_License_Agreement

### CODE REVIEW

**Review by mikelid — APPROVED** [review]
LGTM. Thank you for these improvements.

**Review by Shreyas-vgr — COMMENTED** [review]
Also, please make sure you follow [conventional commit guidelines](https://www.conventionalcommits.org/en/v1.0.0/) for the commit messages since we autogenerate Changelogs using commits. It needs to be in a specific format.

File: doc/userGuide.md [hunk 1]
```diff
@@ -2442,7 +2411,33 @@ this.categoryControl = new ListControl(
...
-# 7. Reference
+# 7. Contributing
+
+## 7.1. Get the source code
+
+To get the Controls Framework source code for perusal or for development, clone the ASK-SDK v2 github repo, and locate the Controls Framework project:
+
+
+```
+git clone https://github.com/alexa/alexa-skills-kit-sdk-for-nodejs/cd alexa-skills-kit-sdk-for-nodejs/ask-sdk/ask-sdk-controls
```
[the remark this answers is not in the document]
Comment (daltonhuynh) [reply doc/userGuide.md]: Fixed. I've updated this section with latest text.

File: doc/userGuide.md [hunk 2]
```diff
@@ -2442,7 +2411,33 @@ this.categoryControl = new ListControl(
...
+npm install -S ask-sdk-controls
+```
+
+## 7.2. Run the Controls Framework regression tests
+
+The regression tests included in the Controls Framework are a good resource for learning. One way to use them is to step through individual tests inside Visual Studio Code:
+
+1. Clone the framework code.
+2. Copy the launch.json from `./ide/vscode/launch.json` to `./vscode`.
+3. Find an interesting test, mark it as `test.only()` then step into it with `F5` (Debug: Start Debugging)
```
[the remark this answers is not in the document]
Comment (daltonhuynh) [reply doc/userGuide.md:2430]: Not too familiar with features, but definitely feel free to edit and add!

File: README.md [hunk 3]
```diff
@@ -35,4 +35,5 @@ This SDK is distributed under the Apache License, Version 2.0, see LICENSE for m
 
 ## Got Feedback?
 
-Request and vote for Alexa features [here](https://alexa.uservoice.com/forums/906892-alexa-skills-developer-voice-and-vote/filters/top?category_id=322783)!
+* [Connect with us on Slack](https://amazonalexa.slack.com/archives/C018QMP3K33)!
+* Request and vote for Alexa features [here](https://alexa.uservoice.com/forums/906892-alexa-skills-developer-voice-and-vote/filters/top?category_id=322783)!
```
[the remark this answers is not in the document]
Comment (daltonhuynh) [reply README.md:39]: Yeah this filters to the ASK SDK category, since UserVoice includes many others outside even of ASK

**Comment by daltonhuynh** [comment]
> Also, please make sure you follow [conventional commit guidelines](https://www.conventionalcommits.org/en/v1.0.0/) for the commit messages since we autogenerate Changelogs using commits. It needs to be in a specific format.

Sorry, not too familiar with how this format works. Do I need to squash my commits and it takes the latest? Please feel free to also just commit these as a single patch and close this PR.

**Comment by Shreyas-vgr** [comment]
> > Also, please make sure you follow [conventional commit guidelines](https://www.conventionalcommits.org/en/v1.0.0/) for the commit messages since we autogenerate Changelogs using commits. It needs to be in a specific format.
> 
> Sorry, not too familiar with how this format works. Do I need to squash my commits and it takes the latest? Please feel free to also just commit these as a single patch and close this PR.

Yes, generally the commit messages should follow the guidelines. Yes, you can squash all the commits since most of them are fixes on UserGuide and have a single commit as following.

`doc: updated the userGuide sections and added slack discussion group links for developers`

**Review by Shreyas-vgr — APPROVED** [review]

**Pull request merged**